docs/secrets.md

```diff
@@ -0,0 +1,62 @@
+# Secrets
+
+<blockquote class="info">
+This article explains how to use secrets in a workspace. To authenticate the
+workspace provisioner, see <a href="./templates/authentication">this</a>.
+</blockquote>
+
+Coder is open-minded about how you get your secrets into your workspaces.
+
+## Wait a minute...
+
+Your first stab at secrets with Coder should be your local method.
+You can do everything you can locally and more with your Coder workspace, so
+whatever workflow and tools you already use to manage secrets may be brought
+over.
+
+For most, this workflow is simply:
+
+1. Give your users their secrets in advance
+1. Your users write them to a persistent file after
+   they've built their workspace
+
+<a href="./templates#parameters">Template parameters</a> are a dangerous way to accept secrets.
+We show parameters in cleartext around the product. Assume anyone with view
+access to a workspace can also see its parameters.
```

Comment:
```suggestion
For most cases, we do not recommend using <a href="./templates#parameters">template parameters</a> to accept workspace secrets as anyone with view access to a workspace can also see its parameters.
```

Reasoning:

1. Workspace params do not show up in the dashboard, CLI, or build log unless the template definition explicitly uses a parameter as label for a resource/coder_app. The same could be done with "secrets"
  - They may be fetchable via the API (cc @Emyrk)
1. The only role that has read access to workspaces also has write access and can enter a terminal and `env` or `cat project/.env` to expose secrets

Reply:
> Workspace params do not show up in the dashboard, CLI, or build log unless the template definition explicitly uses a parameter as label for a resource/coder_app. The same could be done with "secrets"

This is the case right now, but could change in the near future, so I left out the specifics.

> The only role that has read access to workspaces also has write access and can enter a terminal and env or cat project/.env to expose secrets

The specific permission nodes seem likely to change in the near future as we build out RBAC. So, I phrased it in the safest way.